reana_client/cli/test.py

```diff
@@ -0,0 +1,144 @@
+        """Get logs from a workflow engine, use existing API function."""
+        return get_workflow_logs(workflow, self.access_token, steps, page, size)
+
+    def get_workflow_status(self, workflow):
+        """Get of a previously created workflow."""
+        return get_workflow_status(workflow, self.access_token)
+
+    def get_workflow_specification(self, workflow):
+        """Get specification of previously created workflow."""
+        return get_workflow_specification(workflow, self.access_token)
+
+    def download_file(self, workflow, file_path):
+        """Download the requested file if it exists."""
+        return download_file(workflow, file_path, self.access_token)
+
+
+@click.group(help="Workflow run test commands")
+def test_group():
+    """Workflow run test commands."""
+
+
+@test_group.command("test")
+@click.option(
+    "-n",
+    "--test-files",
+    multiple=True,
+    default=None,
+    help="Gherkin file for testing properties of a workflow execution. Overrides files in reana.yaml if provided.",
+)
+@click.pass_context
+@add_access_token_options
+@check_connection
+@add_workflow_option
+def test(ctx, workflow, test_files, access_token):
+    r"""
+    Test workflow execution, based on a given Gherkin file.
+
+    Gherkin files can be specified in the reana specification file (reana.yaml),
+    or by using the ``-n`` option.
+
+    The ``test`` command allows for testing of a workflow execution,
+    by assessing whether it meets certain properties specified in a
+    chosen gherkin file.
+
+    Example:
+        $ reana-client test -w myanalysis -n test_analysis.feature
+        $ reana-client test -w myanalysis
+        $ reana-client test -w myanalysis -n test1.feature -n test2.feature
+    """
+    try:
+        workflow_status = get_workflow_status(
+            workflow=workflow, access_token=access_token
+        )
+        status = workflow_status["status"]
+        id = workflow_status["id"]
+    except Exception:
+        display_message(f"Could not find workflow ``{workflow}``.", msg_type="error")
+        sys.exit(1)
+
+    if status != "finished":
+        display_message(
+            f"``{workflow}`` is {status}. It must be finished to run tests.",
+            msg_type="error",
+        )
+        sys.exit(1)
+
+    if not test_files:
+        reana_specification = get_workflow_specification(workflow, access_token)
+        try:
+            test_files = reana_specification["specification"]["tests"]["files"]
+        except KeyError:
+            display_message(
+                "No test files specified in reana.yaml and no -n option provided.",
+                msg_type="error",
+            )
+            sys.exit(1)
+
+    data_fetcher = DataFetcherClient(access_token)
+    for test_file in test_files:
+        click.secho(f"\nUsing test file {test_file}", fg="cyan", bold=True)
+        try:
+            results = parse_and_run_tests(id, test_file, workflow, data_fetcher)
```

Comment:
The workflow ID is being passed as the analysis run ID, but analysis runs are  things specific to continuous-reuse (see also the related comment in reana-commons).

Currently, what the user has passed as `workflow` is what is being used as an identifier to gather details from REANA API. This can create some problems if the user specifies the workflow name without the run number (e.g. `my-roofit` instead of `my-roofit.2.1`), as different calls to the REANA API might return details of different workflows (e.g. if the user has spawned another `my-roofit` workflow in the meantime).  Should we pass either the full name (that we can get from the response of `get_workflow_status`) or the ID of the workflow as the `workflow` parameter of `parse_and_run_tests`? 

(we can discuss this live given that `workflow` is quite ambiguous in this context)